doc/migrations/4.0.1_4.x.x.md

```diff
@@ -6,3 +6,46 @@ Currently, only the `urn:xmpp:http:upload:0` XMLNS is served.
+
+* [`mod_event_pusher_push.virtual_pubsub_hosts`](../../modules/mod_event_pusher_push/#modulesmod_event_pusher_pushvirtual_pubsub_hosts): the default value was changed to `[]`
+
+* [`mod_event_pusher_sns.presence_updates_topic`](../../modules/mod_event_pusher_sns/#modulesmod_event_pusher_snspresence_updates_topic): default not set
+
+* `mod_event_pusher_sns.muc_host`: the option was removed
+
+* [`mod_global_distrib.connections.advertised_endpoints`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionsadvertised_endpoints): default not set (`false` is no longer accepted)
+
+* `mod_global_distrib.connections.tls.enabled`: flag was removed, TLS is enabled by providing [`cacertfile`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionstlscacertfile) and [`certfile`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionstlscertfile) options
+
+* [`mod_http_upload.s3`](../../modules/mod_http_upload/#s3-backend-options): the section is mandatory
+
+* [`mod_mam_meta.user_prefs_store`](../../modules/mod_mam/#modulesmod_mam_metauser_prefs_store): `false` value is no longer allowed
+
+* [`mod_muc_light.config_schema`](../../modules/mod_muc_light/#modulesmod_muc_lightconfig_schema): minor config fields simplification
+
+* `mod_muc_log`: change of default values for [`css_file`](../../modules/mod_muc_log/#modulesmod_muc_logcss_file) and [`top_link`](../../modules/mod_muc_log/#modulesmod_muc_logtop_link)
+
+* `mod_ping`: [`ping_interval`](../../modules/mod_ping/#modulesmod_pingping_interval) and [`ping_req_timeout`](../../modules/mod_ping/#modulesmod_pingping_req_timeout) explicitly said to be in seconds
+
+* [`mod_push_service_mongoosepush.api_version`](../../modules/mod_push_service_mongoosepush/#modulesmod_push_service_mongoosepushapi_version): the value is now limited to be `"v2"` or `"v3"`
+
+* `mod_stream_management`: minor adjustments of [`buffer_max`](../../modules/mod_stream_management/#modulesmod_stream_managementbuffer_max) and [`ack_freq`](../../modules/mod_stream_management/#modulesmod_stream_managementack_freq) options, [`buffer`](../../modules/mod_stream_management/#modulesmod_stream_managementbuffer) and [`ack`](../../modules/mod_stream_management/#modulesmod_stream_managementack) booleans were added
+
+* [`listen.c2s.tls.ciphers`](../../advanced-configuration/listen/#listenc2stlsciphers), [`listen.http.tls.ciphers`](../../advanced-configuration/listen/#listenhttptlsciphers) and [`outgoing_pools.*.*.connection.tls.ciphers`](../../advanced-configuration/outgoing-connections/#outgoing_poolsconnectiontlsciphers): the ciphers should now be formatted as a specification string
+
+* [listen.http.handlers.mod_websockets.ping_rate](../../advanced-configuration/listen/#listenhttphandlersmod_websocketsping_rate): `none` value is no longer allowed
+
+* `outgoing_pools.generic`: the option was removed
```

Comment:
This pool type was removed as it was an internal setting, not supposed to be set in the configuration file.